core/src/dsnp/api_types.rs

```diff
@@ -0,0 +1,96 @@
+#![allow(dead_code)] // todo: remove after usage
+use crate::dsnp::{dsnp_types::DsnpId, encryption::EncryptionBehavior};
+
+/// Raw page of Graph (or Key) data
+pub type Page = Vec<u8>;
+
+pub struct KeyPair<E: EncryptionBehavior> {
```

Comment:
I'm not sure how having this as a generic might work from outside Rust boundary. I guess we can change it later if it caused problem.